src/Client/Web/Bit.Client.Web.BlazorUI/Components/Navs/BitNav.razor.cs

```diff
@@ -1,16 +1,43 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Components.Routing;
 
 namespace Bit.Client.Web.BlazorUI
 {
-    public partial class BitNav
+    public partial class BitNav : IDisposable
     {
         /// <summary>
         /// The key of the nav item selected by caller
         /// </summary>
-        [Parameter] public string? SelectedKey { get; set; }
+        private string? selectedKey;
+
+        [Inject] private NavigationManager navigationManager { get; set; }
+
+        public bool SelectedKeyHasBeenSet { get; set; }
+
+        [Parameter]
+        public string SelectedKey
+        {
+            get => selectedKey;
+            set
+            {
+                if (value == selectedKey) return;
+                selectedKey = value;
+                SelectedKeyChanged.InvokeAsync(selectedKey);
+
+                var currentUrl = NavLinkItems.Where(nli => nli.Links != null).SelectMany(sli => sli.Links).FirstOrDefault(sli => sli.Key.Contains(selectedKey))?.Key ?? string.Empty;
```

Comment:
please format this line into multiple lines

Reply:
formatted into multiple lines